// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "com.dumselfflydev.chatbot";
option java_outer_classname = "ChatbotProto";
option objc_class_prefix = "HLW";

package chatbot;

// The greeting service definition.
service Chatboter {
    rpc CheckHealth (HealthCheckRequest) returns (HealthCheckResponse) {}
    rpc TrainModel (TrainModelRequest) returns (TrainModelResponse) {}
    rpc GetResponse (GetResponseRequest) returns (GetResponseResponse) {}
}

// The request message containing the user's name.
message HealthCheckRequest {
  string name = 1;
}

// The response message containing the greetings
message HealthCheckResponse {
  string message = 1;
}

// The request message containing the user's name.
message TrainModelRequest {
  repeated string conversation = 1;
}

// The response message containing the greetings
message TrainModelResponse {
  string message = 1;
  int32 status = 2;
}

// The request message containing the user's name.
message GetResponseRequest {
  string message = 1;
}

// The response message containing the greetings
message GetResponseResponse {
  string message = 1;
  int32 status = 2;
}
